api/src/main/java/org/apache/iceberg/transforms/PartitionSpecVisitor.java

```diff
@@ -26,20 +26,94 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
 public interface PartitionSpecVisitor<T> {
-  T identity(String sourceName, int sourceId);
+  default T identity(int fieldId, String sourceName, int sourceId) {
+    return identity(sourceName, sourceId);
+  }
+
+  default T identity(String sourceName, int sourceId) {
+    throw new UnsupportedOperationException("Identity transform is not supported");
+  }
+
+  default T bucket(int fieldId, String sourceName, int sourceId, int width) {
```

Comment:
`width` -> `numBuckets`? Same as in `SortOrderVisitor`L34

Reply:
Fixed.